mlflow/tracing/fluent.py

```diff
@@ -148,7 +148,7 @@ def start_span(
     name: str = "span",
     span_type: Optional[str] = SpanType.UNKNOWN,
     attributes: Optional[Dict[str, Any]] = None,
-):
+) -> Generator[LiveSpan, None, None]:
```

Comment:
@B-Step62 is this correct?

Reply:
Indeed it can yield `NoOpSpan` too in case of failure. If `Union` makes it hard to read, I think `Generator[Span, None, None]` is enough.